### PR TITLE
Add hyprctl output in json when `-j` flag is used

### PR DESCRIPTION
**Pull request by bazuin-32**
#### Describe your PR, what does it fix/add?
Adds the ability to get `hyprctl` output in json, as requested in #164.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
Not all requests have json output added to them (for example `dispatch` does not), only the ones where I thought json output would be useful. If you'd like me to add it to any of the requests that I left out, let me know and I can do that.

#### Is it ready for merging, or does it need work?
I think its ready, as long as it is acceptable. Seems to work well for me.

### CODE REVIEW

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -13,95 +13,386 @@
...
+            result += "\n\n";
         }
-        result += "\n\n";
     }
 
     return result;
 }
 
-std::string devicesRequest() {
+std::string devicesRequest(bool json) {
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:274]: to every single with bool json.

NO.

Make an enum with modes. Name it for example eHyprctlOutputFormat

and make it
FORMAT_NORMAL = 0,
FORMAT_JSON

then pass that.

**Comment by bazuin-32** [comment]
Got it, I'll make those changes

**Comment by bazuin-32** [comment]
Not sure if that's what you meant, let me know if it is or if I missed something.

**Comment by vaxerski** [comment]
I hope you tested everything, including batch, keyword, dispatch, help, etc.

**Comment by bazuin-32** [comment]
Ah whoops, looks like I didn't. I'll have to fix some things then

**Comment by bazuin-32** [comment]
Ok so I tested everything except hyprpaper since I don't have that running. It all works as expected.

**Comment by vaxerski** [comment]
hyprpaper seems to work for me. Everything else tested and ready?

**Comment by bazuin-32** [comment]
Still working on batch, sorry it's taking so long

**Comment by vaxerski** [comment]
no hurries.

**Comment by bazuin-32** [comment]
Ok, I think its finally good. As long as hyprpaper works for you, the rest works fine for me.

**Comment by vaxerski** [comment]
lgtm, thank you!

**Pull request merged**